### PR TITLE
Fix issues on deployment of 7.8.0

### PR DESCRIPTION
**Pull request by matthewrmshin**
As you do... Found the first set of issues on deployment of 7.8.0 on site.

Cylc Review on WSGI - may not have HOME environment variable, so it was failing when attempting to use the `cylc.cfgspec.globalcfg` module to load the `global.rc`. Use `pwd` module instead via `cylc.hostuserutil` instead.

Modify remaining `python` invocation to use `python2`.

### CODE REVIEW

File: lib/cylc/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -414,15 +414,16 @@ class GlobalConfig(ParsecConfig):
...
     # For working with git clones, just go to minor version number.
     cylc_version = re.sub('-.*', '', CYLC_VERSION)
     CONF_BASE = "global.rc"
     # Site global.rc loc preference: if not in etc/ look in old conf/.
     SITE_CONF_DIR = os.path.join(os.environ["CYLC_DIR"], "etc")
     SITE_CONF_DIR_OLD = os.path.join(os.environ["CYLC_DIR"], "conf")
     # User global.rc loc preference: if not in .cylc/x.y.z/ look in .cylc/.
-    USER_CONF_DIR_1 = os.path.join(os.environ['HOME'], '.cylc', cylc_version)
-    USER_CONF_DIR_2 = os.path.join(os.environ['HOME'], '.cylc')
+    USER_CONF_DIR_1 = os.path.join(_HOME, '.cylc', cylc_version)
```
Comment (oliver-sanders) [inline lib/cylc/cfgspec/globalcfg.py]: Would it not have been easier to use `$HOME` and replace it only when needed?

Comment (matthewrmshin) [reply lib/cylc/cfgspec/globalcfg.py]: That's how _HOME is calculated above.

**Comment by kinow** [comment]
Code looks good, just trying to try it out on a WSGI server :+1:

**Comment by kinow** [comment]
Hmmm, do I need WSGI + Apache in order to reproduce the issue?

Running from `master` branch, but decided to try with Nginx https://github.com/kinow/cylc-docker/tree/master/cylc-review

```bash
host$ docker-compose up --build
host$ docker exec -t -i cylcreview_backend_1 /bin/bash
```

```bash
docker$ cylc register five /opt/cylc/etc/examples/tutorial/cycling/five/
docker$ cylc run --no-detach --debug five
```

And while that suite was running, I could successfully search by my user, list my suites, choose one, choose one task, and look at error logs.

**Comment by matthewrmshin** [comment]
Most likely because our Apache web server runs in a very restricted environment.

**Comment by matthewrmshin** [comment]
@hjoliver could you sanity check this one - as @oliver-sanders is away?

**Comment by hjoliver** [comment]
@matthewrmshin - (somewhat :sheep:-ishly :grin: ) diff coverage is at 72.72% (interesting number!) and most the changes would be quite amenable to unit tests...

**Pull request merged**